src/core/unit.c

```diff
@@ -6626,6 +6628,41 @@ static uint64_t unit_get_cpu_weight(Unit *u) {
         return cc ? cgroup_context_cpu_weight(cc, manager_state(u->manager)) : CGROUP_WEIGHT_DEFAULT;
 }
 
+void unit_get_exec_quota_stats(Unit *u, ExecContext *c, ExecDirectoryType dt, uint64_t *usage, uint64_t *limit) {
+        *usage = 0;
+        *limit = 0;
+
+        if (c->directories[dt].n_items > 0) {
+                _cleanup_close_ int fd = -EBADF;
+                _cleanup_free_ char *p = NULL, *pp = NULL;
+
+                ExecDirectoryItem *i = &c->directories[dt].items[0];
+                p = path_join(u->manager->prefix[dt], i->path);
+                if (!p)
+                        return;
```

Comment:
you should log about errors or return an error code. just eating up oom like this is not ok.